src/main/java/com/gorlah/kappabot/function/BotRequestEndpoint.java

```diff
@@ -0,0 +1,12 @@
+package com.gorlah.kappabot.function;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+public enum BotRequestEndpoint {
```

Comment:
IDK if this is going to work out as an enum in the long run. Say for example if you had a plugin which added IRC integration, that plugin would have to use one of these two or require a patch to this repository in order to implement the `BotRequestMetadata`.